compiler/passes/checkResolved.cpp

```diff
@@ -139,6 +139,14 @@ isDefinedAllPaths(Expr* expr, Symbol* ret, RefSet& refs)
                arg->intent == INTENT_REF))
             return 1;
 
+          // For arrays/domains/distributions, we just converted all 'ref' args
```

Comment:
When the new constructor story is fully-implemented, definition will occur when the variable is initialized.  In that case it is moot whether assignment is counted as a definition.  As I stated today, we (the compiler, that is) should not know anything more about the '=' operator than that it is a function.  It is entirely up to the implementing class whether it actually takes 2 arguments and what are the intents of those arguments.
So, I accept "const in" in connection with record-wrapped types as a workaround.  Eventually, the need for this workaround will go away.  "const in" seems right for arguments of that type, in any case.